### ISSUE TITLE
[NEXT-1166] Modal Interception does not work for (.)[dynamic] routes

### ISSUE THREAD

**Issue by khuezy** [vercel/next.js#49614]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.14.0
      npm: 9.3.1
      Yarn: 1.22.19
      pnpm: 7.27.0
    Relevant packages:
      next: 13.4.2-canary.4
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/youthful-butterfly-l44fsy

### To Reproduce

1. Go to sandbox and run the dev server.
2. Click on the "Photos" link
3. Click on "Go to /photos/[author]/[id]", where the author and id is "john/123"
4. The interception does not work, the [author] params is "(.)john"
5. Reload the page, the [author] params is "john"

Now go back to the root and click on the "Static" link. This route doesn't use dynamic routes and the interception works as expected.

### Describe the Bug

When using dynamic routes in the modal, the interception of the current route does not work.
That is, when I'm on the "/photos" page, I want a modal-ize "/photos/[author]/[id]"

### Expected Behavior

When on the "/photos" page, clicking on a child route should intercept the child modal.

<sub>[NEXT-1166](https://linear.app/vercel/issue/NEXT-1166/modal-interception-does-not-work-for-[dynamic]-routes)</sub>

**Comment by abusada** [comment]
I've been facing similar issue, it seems that using: 
> (...) to match segments from the root app directory.

Fixes the issue, see here https://codesandbox.io/p/sandbox/quizzical-cloud-oj5kkw

**Comment by khuezy** [comment]
@abusada Thanks! That should unblock me for now.

**Comment by khuezy** [comment]
Is it really a duplicate? The title says that (...)[dynamic] does not work, but it works in latest canary.
The issue here is that (.)[dynamic] doesn't work

**Comment by khuezy** [comment]
I'm getting errors on rsc using (...) :/

**Comment by Nithur-M** [comment]
> Looks like a duplicate of #48143.
> 
> Using the (...) as a workaround only helps if your root segment is not dynamic.

Yes, any updates on this one? 
I am using dynamic routes as my root and facing this issue.

**Comment by Nithur-M** [comment]
> I've been facing similar issue, it seems that using:
> 
> > (...) to match segments from the root app directory.
> 
> Fixes it. see here https://codesandbox.io/p/sandbox/quizzical-cloud-oj5kkw

Hello, @abusada is it possible to use intercepting routes without using a folder to bind everything? I just want to display all the authors on the homepage without using `/photo` route

**Comment by Nickman87** [comment]
I am having a similar issue, repo for testing is here: https://codesandbox.io/p/sandbox/nextjs-intercepting-routes-s9j6mo?file=%2Fpackage.json%3A17%2C29

Checked with latest canary (`13.4.5-canary.3`) and still present
@feedthejim

**Comment by thanhhuy264** [comment]
I also had the same problem:
Before I used folder [lang] to wrap: work
![image](https://github.com/vercel/next.js/assets/61443868/82065bb6-9548-4a47-889b-d35c8d3d4697)

After I use folder [lang] to wrap: not work
![image](https://github.com/vercel/next.js/assets/61443868/f1305415-21d5-4242-bc1d-aaa003ef50bf)

**Comment by thomasschwinn** [comment]
I have the same problem, with dynamic  the root folder, intercepting does not work

**Comment by MafinBoy** [comment]
I don't think this got completely fixed by #51526
It only fixed it if you have nested dynamic routes like `@modal/(.)[user]/[id]/page.tsx`.
Something like `@modal/(.)[author]/page.tsx` still doesn't work.
Code sandbox forked from OP's example: https://codesandbox.io/p/sandbox/trusting-cray-fxlqlp?file=%2Fpackage.json%3A13%2C30

Should I create a new Issue?

**Comment by tuandev777** [comment]
> I also had the same problem:
> 
> # Before I used folder [lang] to wrap: work
> ![image](https://user-images.githubusercontent.com/61443868/243178588-82065bb6-9548-4a47-889b-d35c8d3d4697.png)
> 
> # After I use folder [lang] to wrap: not work
> ![image](https://user-images.githubusercontent.com/61443868/243178766-79d76c61-f64b-4ee0-a77f-093c0ab0f3ce.png) Any fix this?

I got the same error with locale, have anyone try it yet ?

**Comment by Netail** [comment]
Facing the same issue with the following structure;

```
.
└── [locale]/
    ├── (program)/
    │   ├── programma/
    │   │   └── [slug]/
    │   │       └── [id]/
    │   │           └── page.jsx
    │   └── program/
    │       └── [slug]/
    │           └── [id]/
    │               └── page.jsx
    ├── @modal/
    │   ├── (.)programma/
    │   │   └── [slug]/
    │   │       └── [id]/
    │   │           └── page.jsx
    │   └── (.)program/
    │       └── [slug]/
    │           └── [id]/
    │               └── page.jsx
    └── layout.jsx
```

**Comment by khuezy** [comment]
It might be best to open a new issue w/ the examples.

**Comment by MafinBoy** [comment]
> It might be best to open a new issue w/ the examples.

It turns out there's already #52533
Please forward future discussion to that issue
